src/Features/Core/Portable/Extensions/ExtensionFolder.cs

```diff
@@ -65,16 +64,31 @@ public ExtensionFolder(
                     analyzerAssemblyLoader.AddDependencyLocation(dll);
                 }
 
-                return analyzerAssemblyLoader;
+                return (IAnalyzerAssemblyLoaderInternal?)analyzerAssemblyLoader;
+#else
+                // We only support loading extensions in a .Net Core host.
+                return (IAnalyzerAssemblyLoaderInternal?)null;
+#endif
             });
         }
 
+        public void Unload()
```

Comment:
Should this be a `Dispose` instead?  And also wondering if we should add in any guardrails to throw in the other entrypoints if this has been disposed

Reply:
so Unloading is just a *suggestion* to the runtime.  it has no actual impact.  :)